File: CSharpCore/HamaraBasketTests/MovieTicketsQualityCheckerTests.cs
using HamaraBasketQualityAssurance;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace HamaraBasketTests
{
    [TestClass]
    public class MovieTicketsQualityCheckerTests
    {
        [TestMethod]
        public void QualityIncreasesAsSellByValueApproaches()
        {
            var item = new HamaraBasketItem(20, 10, ItemType.MovieTicket);
            var checker = new MovieTicketsQualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(11, item.Quality);
        }

        [TestMethod]
        public void QualityIncreasesByTwoWhenSellByValueIsLessThan10()
        {
            var item = new HamaraBasketItem(10, 10, ItemType.MovieTicket);
            var checker = new MovieTicketsQualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(12, item.Quality);
        }

        [TestMethod]
        public void QualityIncreasesByThreeWhenSellByValueIsLessThan10()
        {
            var item = new HamaraBasketItem(3, 10, ItemType.MovieTicket);
            var checker = new MovieTicketsQualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(13, item.Quality);
        }

        [TestMethod]
        public void QualityDropsToZeroWhenSellByHasPassed()
        {
            var item = new HamaraBasketItem(0, 10, ItemType.MovieTicket);
            var checker = new MovieTicketsQualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(0, item.Quality);
        }
    }
}

File: Java/src/test/java/com/hamaraBasket/MovieTicketsQualityCheckerTests.java
package com.hamaraBasket;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class MovieTicketsQualityCheckerTests
    {
        @Test
        public void qualityIncreasesAsSellByValueApproaches()
        {
            HamaraBasketItem item = new HamaraBasketItem(20, 10, ItemType.MovieTicket);
            QualityChecker checker = new MovieTicketsQualityChecker();

            checker.updateQuality(item);

            assertEquals(11, item.getQuality());
        }

        @Test
        public void qualityIncreasesByTwoWhenSellByValueIsLessThan10()
        {
            HamaraBasketItem item = new HamaraBasketItem(10, 10, ItemType.MovieTicket);
            QualityChecker checker = new MovieTicketsQualityChecker();

            checker.updateQuality(item);

            assertEquals(12, item.getQuality());
        }

        @Test
        public void qualityIncreasesByThreeWhenSellByValueIsLessThan10()
        {
            HamaraBasketItem item = new HamaraBasketItem(3, 10, ItemType.MovieTicket);
            QualityChecker checker = new MovieTicketsQualityChecker();

            checker.updateQuality(item);

            assertEquals(13, item.getQuality());
        }

        @Test
        public void qualityDropsToZeroWhenSellByHasPassed()
        {
            HamaraBasketItem item = new HamaraBasketItem(0, 10, ItemType.MovieTicket);
            QualityChecker checker = new MovieTicketsQualityChecker();

            checker.updateQuality(item);

            assertEquals(0, item.getQuality());
        }
    }